### DISCUSSION TITLE
Where is the asset with FileUploads & standalone javascript library

### DISCUSSION THREAD

**Discussion by lukadschaak** [dachcom-digital/pimcore-formbuilder#497] [Q&A]
I have multiple questions, so i hope it is okay to write them in one discussion.

### 1

On the [doc site for FileUpload](https://github.com/dachcom-digital/pimcore-formbuilder/blob/master/docs/80_FileUpload.md), it qoutes "Different storage principals: Store data as pimcore assets (/formdata asset folder) and add download-link to mail or add them as native mail attachments"

Does that mean files are not storred to `formdata` if no mail is sended?

I have a working form with file upload and the file ist storred under `var/tmp/form-builder-files`. But no asset is created unter asset folder `formdata`.

Am i missing some configuration? I mean i could use the `FormEvents::POST_SUBMIT` event to do it manually, but is that the way to go? If yes, more docs for this would maybe be helpful.

### 2

For forms that require javascript (in my case file uploads), i can choose from a javascript package or an jquery package. But both is only installable via npm (refering to [Javascript Plugins](https://github.com/dachcom-digital/pimcore-formbuilder/blob/master/docs/91_Javascript.md)).

Is there no way of providing standalone javascript without the need of npm? Would be that be the `Alpine.js` with the TODO section the solution for that?

Because in the current way the formbuilder seems not compatible with raw Pimcore projects, that only uses Twig right?

**Comment by lukadschaak** [comment]
no one?

**Comment by solverat** [comment]
## Asset Management

> Does that mean files are not storred to formdata if no mail is sended?

If you haven't selected the `Send Files as Attachment` checkbox in the upload field config, files will be uploaded asynchronously to the tmp directory (therefor you need to implement one of the JavaScript libraries or implement the upload process by yourself). After the form gets submitted, the files will be packed as a zip file and transformed to a pimcore asset. 

If you haven't checked the checkbox, the workflow stays the same, but the files will be attached to the emails (configured in workflow channels) only (no pimcore asset will be generated).

## Libraries
No, there has to be a build process in some way (webpack/encore/gulp). Even with alpine.js, this wouldn't change. 
We also develop within the good ol' twig ecosystem sometimes, a proper JS build is still mandatory (these days). So of course these extensions support projects with a classic twig setup.

However, that's why we decided to decouple the JS extensions from the core repo: to allow users to write their own custom FE implementation.
